src/Timeline.cpp

```diff
@@ -157,6 +174,49 @@ namespace gamescope
 
             return { nExplicitSyncEventFd, false };
         }
+
+    }
+
+    template <TimelinePointType Type>
+    std::shared_ptr<VulkanSemaphore_t> CTimelinePoint<Type>::ToVkSemaphore()
+    {
+        uint32_t uSyncobjHandle = m_pTimeline->GetSyncobjHandle();
+        uint32_t uTmpSyncobj = 0;
+        int nSyncFd = -1;
+
+        if ( drmSyncobjCreate( g_device.drmRenderFd(), 0, &uTmpSyncobj ) != 0 )
+        {
+            goto done_ToVkSemaphore;
+        }
+
+        if ( drmSyncobjTimelineWait(
+             m_pTimeline->GetDrmRenderFD(),
+             &uSyncobjHandle,
+             &m_ulPoint,
+             1,
+             std::numeric_limits<int64_t>::max(),
+             DRM_SYNCOBJ_WAIT_FLAGS_WAIT_AVAILABLE,
+             nullptr ) != 0 )
+        {
+            goto done_ToVkSemaphore;
+        }
+
+        if ( drmSyncobjTransfer( g_device.drmRenderFd(),
+                                 uTmpSyncobj, 0, //dst
+                                 uSyncobjHandle, m_ulPoint,
+                                 0 ) != 0 )
+        {
+            goto done_ToVkSemaphore;
+        }
+
+        drmSyncobjExportSyncFile( g_device.drmRenderFd(),
+                                  uTmpSyncobj,
+                                  &nSyncFd );
+
+done_ToVkSemaphore:
+        drmSyncobjDestroy( g_device.drmRenderFd(), uTmpSyncobj );
+
+        return g_device.ImportSyncFd( nSyncFd );
```

Comment:
This will skip the synchronization (I.e., it creates a VkSemaphore in the signaled state from my understanding) if there are any errors in this function. Is that the intent?